Bug with the LlamaCpp logits processor argument

While working on the integration with Pydantic AI, I ran into the following error while using the `LlamaCpp` model with `output_type=None`:

```python
E                     File "/home/robinpicard/pydantic-ai/.venv/lib64/python3.12/site-packages/llama_cpp/llama.py", line 709, in apply_func
E                       recarray.logit[:] = logit_processor(self._input_ids, recarray.logit)
E                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E                   TypeError: 'NoneType' object is not callable
```

It's surprising the issue does not arise in all cases, but looking at the `llama_cpp_python` code base, it appears that our current implementation is faulty: we should provide `None` as the value for the `logits_processor` argument if there's no output type instead of `LogitsProcessorList([None])`

--- a/outlines/models/llamacpp.py
+++ b/outlines/models/llamacpp.py
@@ -197,7 +197,7 @@ class LlamaCppTypeAdapter(ModelTypeAdapter):
 
     def format_output_type(
         self, output_type: Optional[OutlinesLogitsProcessor] = None,
-    ) -> "LogitsProcessorList":
+    ) -> Optional["LogitsProcessorList"]:
         """Generate the logits processor argument to pass to the model.
 
         Parameters
@@ -213,7 +213,9 @@ class LlamaCppTypeAdapter(ModelTypeAdapter):
         """
         from llama_cpp import LogitsProcessorList
 
-        return LogitsProcessorList([output_type])
+        if output_type is not None:
+            return LogitsProcessorList([output_type])
+        return None
 
 
 class LlamaCpp(Model):

